pkg/bundle/sync.go

```diff
@@ -162,6 +200,17 @@ func (b *bundle) syncTarget(ctx context.Context, log logr.Logger,
 	var configMap corev1.ConfigMap
 	err := b.targetDirectClient.Get(ctx, client.ObjectKey{Namespace: namespace.Name, Name: bundle.Name}, &configMap)
 
+	if target.AdditionalFormats != nil && target.AdditionalFormats.JKS != nil {
+		if target.AdditionalFormats.JKS.Key == target.ConfigMap.Key {
+			return false, errors.New("invalid target (duplicate key)")
+		}
```

Comment:
suggestion: I should've suggested this before, my bad - maybe we could move this check to the webhook? (https://github.com/cert-manager/trust-manager/blob/main/pkg/webhook/validation.go)

That would mean that when users make the change it'll error, rather than when we attempt to Sync it.

Reply:
Yep - that's a better place for it. :)